crates/rpc-types-eth/src/filter.rs

```diff
@@ -215,12 +215,32 @@ impl FilterBlockOption {
     /// Sets the block number this range filter should start at.
     #[must_use]
     pub fn with_from_block(&self, block: BlockNumberOrTag) -> Self {
+        // Check if from_block is greater than to_block
+        if self
+            .get_to_block()
+            .and_then(|to| to.as_number())
+            .zip(block.as_number())
+            .map_or(false, |(to, from)| from > to)
+        {
+            panic!("from_block cannot be greater than to_block");
+        }
```

Comment:
hmm, we should not panic here.

worst thing that can happen when sending an invalid range is an rpc error as response

Reply:
Yes true, didn't think about this properly